### PR TITLE
Add logic for building navigation to Publisher.

### PR DESCRIPTION
**Pull request by samiwel**
### What is the context of this PR?
The Publisher was not building the navigation correctly for the final Survey Runner JSON.
This PR adds the logic to build the navigation JSON and adds some tests to ensure correct behaviour.

### How to review 
The schema generated using Publish should contain navigation JSON block containing the section titles.

### CODE REVIEW

**Comment by samiwel** [comment]
[Build navigation in Publisher](https://trello.com/c/pq1jolEi/272-build-navigation-in-publisher)

File: src/eq_schema/Questionnaire.js [hunk 1]
```diff
@@ -13,13 +14,29 @@ class Questionnaire {
...
     return groups;
   }
+
+  buildNavigation(visible, sections) {
+    return {
+      visible,
+      sections: sections.map(section => {
+        return {
+          title: getInnerHTML(section.title),
+          group_order: [`group-${section.id}`]
```
Comment (WickyNilliams) [inline src/eq_schema/Questionnaire.js]: out of curiosity, what does survey runner do with this?

Comment (samiwel) [reply src/eq_schema/Questionnaire.js]: I'm not entirely sure why it needs this, but on first glance it uses the `group_order` for checking if the group is to be skipped or not, and for determining which section should be highlighted. Perhaps @ajmaddaford can provide some additional context?

File: src/eq_schema/Questionnaire.js [hunk 2]
```diff
@@ -13,13 +14,29 @@ class Questionnaire {
...
     last(groups).blocks.push(new Summary());
     return groups;
   }
+
+  buildNavigation(visible, sections) {
+    return {
+      visible,
+      sections: sections.map(section => {
+        return {
+          title: getInnerHTML(section.title),
```
[the remark this answers is not in the document]
Comment (samiwel) [reply src/eq_schema/Questionnaire.js]: Yeah good point, will have a look to see how Runner handles this today.

**Comment by WickyNilliams** [comment]
Tested with [this questionnaire](http://eq-author.apps.devtest.onsclofo.uk/#/eq-author/questionnaire/109/design/186/369). Looks good to me

image

**Pull request merged**